### DISCUSSION TITLE
Should "Hide Live Streams" hide upcoming premieres?

### DISCUSSION THREAD

**Discussion by miangraham** [FreeTubeApp/FreeTube#2850] [General]
The Hide Live Streams option currently hides some-but-not-all upcoming premieres, and I'm not sure what the proper behavior is. Opening a discussion rather than an issue because I can't tell whether it's a bug.

If the option is supposed to hide all future vids, which seems like a reasonable thing to do, this looks like a bug to fix.

If some should remain under the desired behavior and this is working basically as intended, I'd like to add a new option to hide the remaining upcoming videos due to personal annoyance and have some hacking time this week to attempt the addition.

My use case for clarity: When using FreeTube I'm only interested in seeing complete VODs. I'm the Ghost of Videos Past. Present and Future are not my thing.

Channel where it's currently easy to see undesired behavior: https://www.youtube.com/user/CppCon. This is one channel of many that posts premieres months in advance, adding useless noise to the Subscriptions page.

Screenshots below are using `npm run dev` against f4d14ee4046d8fead1eda05ee296b66c7bbc2aaa (cloned today).

#### Appearance in Subscriptions with Hide Live Streams OFF:

![20221110_13h24m45s_grim](https://user-images.githubusercontent.com/704580/201002339-ce0eb64a-8e9b-4cd4-ab9f-2cf573706a4e.png)

6/7 future videos marked as live. The non-live one in the middle premieres in 15 days.

#### Appearance in Subscriptions with Hide Live Streams ON:

![20221110_13h25m25s_grim](https://user-images.githubusercontent.com/704580/201003239-7f1ba6c0-081d-424f-a8a6-3989d2089015.png)

The videos that were already marked live are now hidden, but the non-live upcoming premiere remains visible as the first item in the list. This is the one I'd like to see gone.

Related: https://github.com/FreeTubeApp/FreeTube/issues/507, https://github.com/FreeTubeApp/FreeTube/issues/1070, https://github.com/FreeTubeApp/FreeTube/issues/2755

Should I file a bug and try to fix, or start adding a new option?

**Comment by miangraham** [comment]
While waiting for opinions I'm going to start hacking on this as an independent new option similar to #507's request. If it makes more sense to combine it, I can make that fix before merge.

Please shout back and stop me if there's already similar work in progress somewhere I haven't found on my own.

**Comment by miangraham** [comment]
First pass at adding this as a new option: https://github.com/FreeTubeApp/FreeTube/pull/2853

Thoughts appreciated.

**Comment by ChunkyProgrammer** [comment]
Sorry that no one got back to you on this. Glad to see your change was merged though 🙂
